vignettes/speedup_options.Rmd

```diff
@@ -0,0 +1,710 @@
+library(rstan)
+library(cmdstanr)
+library(ggplot2)
+library(dplyr)
+library(lubridate)
+library(scales)
+library(posterior)
+```
+
+This vignette benchmarks available model options in `{EpiNow2}` and compares the models based on their speed (run time) and accuracy of estimates relative to data with known trajectories. Thoughout, we compare various models to the default model as it is the first model most users encounter.
+
+## The benchmarking data
+
+To compare the model options, we will need a dataset for which we know the "true" values and trajectories. So, we will start by generating the "true" infections and $R_t$ data using `{EpiNow2}`'s `forecast_infections()` function. 
+
+`forecast_infections()` requires a fitted estimates object from `epinow()` with `output` set to "fit", the trajectory of the reproduction number, `R`, and the number of samples to simulate. So, we will set these up first.
+
+To obtain the `estimates` object, we will run the `epinow()` function using real-world observed data and delay distributions to recover realistic parameter values. We will use the first $100$ observations of the `example_confirmed` data set, the `example_generation_time`, `example_incubation_period`, and `example_reporting_delay` values that come with `{EpiNow2}`.
+
+Throughout this vignette, several of the argument values be maintained (not varied), so we will define them here. These include the generation time, incubation period, reporting delay, observation model options, and the forecast horizon.
+
+``` r
+# Set the number of cores to use
+options(mc.cores = 4)
+
+# Generation time
+generation_time <- Gamma(
+  shape = Normal(1.3, 0.3),
+  rate = Normal(0.37, 0.09),
+  max = 14
+)
+
+# Incubation period
+incubation_period <- LogNormal(
+  meanlog = Normal(1.6, 0.05),
+  sdlog = Normal(0.5, 0.05),
+  max = 14
+)
+
+# Reporting delay
+reporting_delay <- LogNormal(
+  meanlog = 0.5,
+  sdlog = 0.5,
+  max = 10
+)
+
+# Combine the incubation period and reporting delay into one delay
+delay <- incubation_period + reporting_delay
+
+# Observation model options
+obs <- obs_opts(
+  scale = list(mean = 0.1, sd = 0.025),
+  return_likelihood = TRUE
+)
+
+# 7-day forecast window
+horizon <- 7
+
+# Rt default prior
+rt_prior_default <- list(mean = 2, sd = 0.1)
+```
+
+Now, let's generate the `estimates` object from `epinow()`.
+
+``` r
+estimates <- epinow(
+  data = example_confirmed[1:100],
+  generation_time = generation_time_opts(example_generation_time),
+  delays = delay_opts(example_incubation_period + example_reporting_delay),
+  rt = rt_opts(prior = rt_prior_default, rw = 14),
+  gp = NULL,
+  obs = obs,
+  horizon = 0, # no forecasting
+  output = "fit"
+)
+#> DEBUG [2024-08-07 14:41:13] epinow: Running in exact mode for 2000 samples (across 4 chains each with a warm up of 250 iterations each) and 114 time steps of which 0 are a forecast
+```
+
+For the `R` data, we will set up an arbitrary trajectory and add some Gaussian noise.
```

Comment:
Suggest expanding on why this Rt trajectory is the way it is. I.e constant periods, periods of linear change and what that might look like in the real world